org.adempiere.base/src/org/compiere/model/MProduct.java

```diff
@@ -698,7 +698,7 @@ protected boolean beforeSave (boolean newRecord)
 		
 		//	Reset Stocked if not Item
 		//AZ Goodwill: Bug Fix isStocked always return false
-		if (!PRODUCTTYPE_Item.equals(getProductType()))
+		if (!PRODUCTTYPE_Item.equals(getProductType()) && !PRODUCTTYPE_ExpenseType.equals(getProductType()))

```

Comment:
please change the comment too.

Also, please check line 666 of MProduct, there is a validation and storage managing code when changing from Item to not-Item, not sure about the effect, probably changing from Item to Expense doesn't matter now if the Stocked flag is maintained.

Line 918 (beforeDelete) has also a validation about storage